qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/plot_presenter.py

```diff
@@ -103,6 +112,70 @@ def do_serial_fit(self):
     def do_seq_fit(self):
         self.fit_all_started_notifier.notify_subscribers(True)
 
+    def _re_organize_keys_find_peaks_convolve(self, table_ws):
+        if table_ws.rowCount() == 1:
+            table_dict = table_ws.row(0)
+            table_dict.pop("SpecIndex", None)
+            return {col_name.split("_")[-1]: value for col_name, value in table_dict.items()}
+        return None
+
+    def run_find_peaks_convolve(self):
+        self.is_waiting_convolve_peaks = False
+        self.find_peaks_convolve_started_notifier.notify_subscribers()
+        try:
+            input_ws_name = self.view.fit_browser.workspaceName()
+            if not ADS.doesExist(input_ws_name):
+                logger.error(f"{input_ws_name} does not exist!")
+                self.find_peaks_convolve_done_notifier.notify_subscribers(False)
+                return
+            fit_prop_ws = ADS.retrieve(input_ws_name)
+            groupWs = FindPeaksConvolve(
```

Comment:
Should `FindPeaksConvolve` be run as async worker as well - like Fit? 
https://github.com/mantidproject/mantid/blob/1d722ac311187af24192a00c3516d0777190de39/qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/plot_presenter.py#L106

Reply:
Since the runtime of FindPeaksConvolve was just in seconds(<1sec) I thought a blocking call would be sufficient. Do you think it is prudent to call algorithms like these in Async mode?

Reply:
Yeh I see, running as async may not be necessary atm,  but I think it would be future proof  - e.g. what happens if data size increases or `FindPeaksConvolve` gets slower because of additional functionality added? If you think it's a bad idea, I'm happy to keep it as is.

Reply:
I tried adapting the Async mode, but it gives this error when the async thread is trying to do the update on the fit browser object which is on the main thread.
```
Exception in thread Thread-8:
Traceback (most recent call last):
 File "C:\Users\kli94267\.conda\envs\mantid-developer\lib\threading.py", line 1016, in _bootstrap_inner
 self.run()
 File "C:\Codes\mantidproject\mantid\mantid\qt\python\mantidqt\mantidqt\utils\asynchronous.py", line 76, in run
 self.finished_cb()
 File "C:\Codes\mantidproject\mantid\mantid\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\fitting\plotting\plot_presenter.py", line 168, in _on_done_find_peaks_convolve
 self.view.fit_browser.loadFunction(self.async_output)
Exception: unknown
```